### ISSUE TITLE
Linker Error while using folly.

### ISSUE THREAD

**Issue by arunt1204** [facebook/folly#1249]
I have built folly in windows machine via "vcpkg install folly:x64-windows" as mentioned in GitHub.

I'm using Visual Studio 2019 c++. The c++ code below,

#include <iostream>
#include <folly/futures/Future.h>
#include <folly/executors/ThreadedExecutor.h>
using namespace folly;
using namespace std;

int main()
{
	folly::ThreadedExecutor executor;
	cout << "making Promise" << endl;
	Promise<int> p; // Inclusion of this stmt throws the below linker error.
        //some code
}

While I build this project using VSC++, it throws the below error,
Error	LNK2019	unresolved external symbol __imp_WSAGetLastError referenced in function "__int64 __cdecl folly::portability::unistd::read(int,void *,unsigned __int64)" (?read@unistd@portability@folly@@YA_JHPEAX_K@Z)	Folly.

Pls help in resolving this linker error.

**Comment by Orvid** [comment]
How are you linking against Folly? It sounds like the windows system libraries folly is dependent on aren't getting linked in.

**Comment by arunt1204** [comment]
I build folly via "vcpkg install folly:x64-windows" cmd.
As folly is build via vcpkg, Visual Studio 2019c++ by its own linking with folly. 

As you mentioned folly dependent windows library aren't getting linked, I have tried adding the path of windows .lib, but the same error occurs.

Below are the list of .lib files (of x64), I linked,
kernel32.lib;user32.lib;gdi32.lib;winspool.lib;comdlg32.lib;advapi32.lib;shell32.lib;ole32.lib;oleaut32.lib;uuid.lib;odbc32.lib;odbccp32.lib;
